crypto/context.c

```diff
@@ -155,11 +169,7 @@ static int openssl_ctx_init_index(OPENSSL_CTX *ctx, int static_index,
     int idx;
 
 #ifndef FIPS_MODE
-    if (ctx == NULL) {
-        if (!RUN_ONCE(&default_context_init, do_default_context_init))
-            return 0;
-        ctx = default_context;
-    }
+    ctx = openssl_ctx_get_concrete(ctx);
```

Comment:
Do we want this kind of pattern? Or would it be better to just call the function always and 
#ifdef FIPS_MODE
# define openssl_ctx_get_concrete(ctx) ctx
#endif